src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -0,0 +1,297 @@
+                
+                default:
+                    throw new Exception($"Got {type} and the value: {field.ValueType} at UnaryMatch.");
+            }
+        }
+
+        private IQueryMatch EvaluateBetween(BetweenExpression betweenExpression, bool negated)
+        {
+            var exprMin = GetValue(betweenExpression.Min);
+            var exprMax = GetValue(betweenExpression.Max);
+            var fieldId = GetFieldIdInIndex(GetFieldName((FieldExpression)betweenExpression.Source));
+            
+            switch (exprMin.ValueType, exprMax.ValueType, negated)
+            {
+                case (ValueTokenType.Long, ValueTokenType.Long, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                case (ValueTokenType.Long, ValueTokenType.Long, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                
+                case (ValueTokenType.Double, ValueTokenType.Double, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (double)exprMin.FieldValue, (double)exprMax.FieldValue);
+                case (ValueTokenType.Double, ValueTokenType.Double, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (double)exprMin.FieldValue, (double)exprMax.FieldValue);
+                
+                case (ValueTokenType.String, ValueTokenType.String, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                case (ValueTokenType.String, ValueTokenType.String, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                
+                
+                default:
+                    var unsupportedType = exprMin.ValueType is ValueTokenType.Double or ValueTokenType.Long or ValueTokenType.String
+                        ? exprMax.ValueType
+                        : exprMin.ValueType;
+                    throw new InvalidDataException($"Got {unsupportedType} but expected: {ValueTokenType.String}, {ValueTokenType.Long}, {ValueTokenType.Double}.");
+            }
+        }
+
+        private IQueryMatch EvaluateInExpression(FieldExpression f, List<QueryExpression> list)
+        {
+            var values = new List<string>();
+            foreach (ValueExpression v in list)
+                values.Add(v.Token.Value);
+
+            return _searcher.InQuery(f.FieldValue, values);
+        }
+
+        private IQueryMatch OrderBy(IQueryMatch match, List<(QueryExpression Expression, OrderByFieldType FieldType, bool Ascending)> orders)
+        {
+            //todo maciej: add support for function in orderby. Also need to implement this as corax method.
```

Comment:
what do you mean function?

Reply:
Add support for MethodExpression. Now I allow only FieldExpression.